Sources/TuistSupport/Utils/FileHandler.swift

```diff
@@ -226,3 +244,61 @@ public class FileHandler: FileHandling {
         }
     }
 }
+
+// Need to find a suitable place for XCFrameworkInfoPlist
+
+public struct XCFrameworkInfoPlist: Codable {
+    
+    enum CodingKeys: String, CodingKey {
+        case libraries = "AvailableLibraries"
+    }
+
+    public struct Library: Codable {
+    
+         enum CodingKeys: String, CodingKey {
+               case identifier = "LibraryIdentifier"
+               case path = "LibraryPath"
+               case architectures = "SupportedArchitectures"
+        }
+
+        public enum Platform: String, Codable {
+            case ios
+        }
+        
+        public let identifier: String
+        public let path: RelativePath
+        public let architectures: [BinaryArchitecture]
+        
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            identifier = try container.decode(String.self, forKey: .identifier)
+            path = try container.decode(RelativePath.self, forKey: .path)
+            architectures = try container.decode([BinaryArchitecture].self, forKey: .architectures)
+        }
+        
+        public func encode(to encoder: Encoder) throws {
+            var container = encoder.container(keyedBy: CodingKeys.self)
+            try container.encode(identifier, forKey: .identifier)
+            try container.encode(path, forKey: .path)
+            try container.encode(architectures, forKey: .architectures)
+        }
+    }
+
+    public let libraries: [Library]
+}
+
+// TODO: Introduce a duplicate`BinaryArchitecture` as other one from TuistCore cannot be used
+// as there is a circular dependency between TuistSupport and TuistCore, will move it to
+// TuistSupport after discussing with Core team.
+
+public enum BinaryArchitecture: String, Codable {
```

Comment:
@tuist/core guys what do you think of moving `BinaryArchitecture` to `TuistSupport`, please refer to aforementioned `TODO` for a context?

Reply:
This can remain within `TuistCore` - the new utility we need within `TuistSupport.FileHander` is a mechanism to obtain read `Data` for a particular `AbsolutePath` - PropertyList decoding can remain local to the `XCFramrworkNode.parse` method.

We can keep things local as there isn't yet a strong need to share plist reading / decoding elsewhere - this can be revisited in the future if more use cases crop up :)

Reply:
`TuistSupport` should not contain any knowledge about Tuist and for that reason I would not take `BinaryArchitecture` down to the support framework.